File: academic-system-python/src/academic_system/security/authentication_service.py
from academic_system.model.user import User
from academic_system.model.role import Role
from academic_system.security.session import Session
from academic_system.exception.exceptions import AuthenticationException
import logging

logger = logging.getLogger(__name__)


class AuthenticationService:
    def __init__(self):
        self._users = [
            User("admin", "admin123", Role.ADMIN),
            User("professor", "prof123", Role.PROFESSOR),
        ]

    def authenticate(self, username: str, password: str) -> User:
        for user in self._users:
            if user.username == username and user.password == password:
                Session.login(user)
                logger.info(f"Successful login: username={username}, "
                            f"role={user.role}")
                return user
        logger.warning(f"Failed login attempt: username={username}")
        raise AuthenticationException("Invalid username or password.")

File: academic-system/src/main/java/org/example/academic/system/security/AuthenticationService.java
package org.example.academic.system.security;

import org.example.academic.system.exception.AuthenticationException;
import org.example.academic.system.model.Role;
import org.example.academic.system.model.User;

import java.util.ArrayList;
import java.util.List;

public class AuthenticationService {

    private List<User> users;

    public AuthenticationService() {
        users = new ArrayList<>();
        // Usuários padrão do sistema
        users.add(new User("admin", "admin123", Role.ADMIN));
        users.add(new User("professor", "prof123", Role.PROFESSOR));
    }

    public User authenticate(String username, String password) {
        for (User user : users) {
            if (user.getUsername().equals(username)
                    && user.getPassword().equals(password)) {
                Session.login(user);
                return user;
            }
        }
        throw new AuthenticationException("Invalid username or password.");
    }
}